lib/puppet/settings/ini_file.rb

```diff
@@ -12,7 +12,7 @@ def self.update(config_fh, &block)
   def self.parse(config_fh)
     config = new([DefaultSection.new])
     config_fh.each_line do |line|
-      case line
+      case line.gsub(/\n$/, '')
```

Comment:
Any reason not to use line.chomp instead?

Reply:
I tried it both ways and worked either way. Not sure if one is more correct than the other?

Reply:
chomp's probably better as there probably isn't a regex involved.

Reply:
@jtappa: [String#chomp](https://ruby-doc.org/core-2.5.0/String.html#method-i-chomp) was the thing I was trying to remember the name of when we were originally talking about this issue. I couldn't remember whether `chomp` was smart enough to not remove the last character if it wasn't a newline. I was originally confusing it with `chop` which does the "always remove, even if it's not a newline" behavior that we want to avoid here.